src/v/archival/ntp_archiver_service.cc

```diff
@@ -2925,9 +2925,10 @@ ss::future<bool> ntp_archiver::do_upload_local(
 
     auto offset = upload.final_offset;
     auto base = upload.starting_offset;
-    auto ot_state = _parent.get_offset_translator_state();
-    auto delta = base - model::offset_cast(ot_state->from_log_offset(base));
-    auto delta_offset_next = ot_state->next_offset_delta(upload.final_offset);
+    auto log = _parent.log();
+    auto delta = base - model::offset_cast(log->from_log_offset(base));
+    auto delta_offset_next = log->delta(
+      model::next_offset(upload.final_offset));
```

Comment:
i guess i need to go back and look closer, but why didn't we preserve `next_offset_delta` interface?

Reply:
Ah it's just syntactic sugar I added some time ago that ended up only having this one caller, so I didn't think it was worth it